### PR TITLE
Revise password blacklist documentation

### PR DESCRIPTION
**Pull request by thomasdarimont**
Depends on: https://github.com/keycloak/keycloak/pull/14411
See: https://github.com/keycloak/keycloak/issues/8982

### CODE REVIEW

**Review by rmartinc — DISMISSED** [review]
Technical details are updated OK for issue https://github.com/keycloak/keycloak/issues/8982. Grammatical or style details are for somebody else... :smile:

File: server_admin/topics/authentication/password-policies.adoc [hunk 1]
```diff
@@ -82,7 +82,14 @@ Password must not be in a blacklist file.
 
 * Blacklist files are UTF-8 plain-text files with Unix line endings. Every line represents a blacklisted password.
 * {project_name} compares passwords in a case-insensitive manner. All passwords in the blacklist must be lowercase.
-* The value of the blacklist file must be the name of the blacklist file.
-* Blacklist files resolve against `${jboss.server.data.dir}/password-blacklists/` by default. Customize this path using:
-** The `keycloak.password.blacklists.path` property.
-** The `blacklistsPath` property of the `passwordBlacklist` policy SPI configuration.
+* The value must be the name of the blacklist file, e.g. `100k_passwords.txt`.
```
Comment (andymunro) [inline server_admin/topics/authentication/password-policies.adoc]:
```suggestion
* The value must be the name of the blacklist file,  for example, `100k_passwords.txt`.
```

File: server_admin/topics/authentication/password-policies.adoc [hunk 2]
```diff
@@ -82,7 +82,14 @@ Password must not be in a blacklist file.
...
+* Blacklist files resolve against `$KC_HOME/data/password-blacklists/` by default. Customize this path using:
+** The `keycloak.password.blacklists.path` system property.
+** The `blacklistsPath` property of the `passwordBlacklist` policy SPI configuration. To configure the blacklist folder via CLI use `--spi-password-policy-password-blacklist-blacklists-path=/path/to/blacklistsFolder`.
```
Comment (andymunro) [inline server_admin/topics/authentication/password-policies.adoc]:
```suggestion
** The `blacklistsPath` property of the `passwordBlacklist` policy SPI configuration. To configure the blacklist folder using the CLI, use `--spi-password-policy-password-blacklist-blacklists-path=/path/to/blacklistsFolder`.
```

File: server_admin/topics/authentication/password-policies.adoc [hunk 3]
```diff
@@ -82,7 +82,14 @@ Password must not be in a blacklist file.
...
+
+A note about False Positives
+
+The current implementation uses a BloomFilter for fast and memory efficient containment checks, e.g. whether a given password is contained in a blacklist, with the possibility for false positives.
```
Comment (andymunro) [inline server_admin/topics/authentication/password-policies.adoc]:
```suggestion
The current implementation uses a BloomFilter for fast and memory efficient containment checks, such as whether a given password is contained in a blacklist, with the possibility for false positives. 
```

File: server_admin/topics/authentication/password-policies.adoc [hunk 4]
```diff
@@ -82,7 +82,14 @@ Password must not be in a blacklist file.
...
+* Blacklist files resolve against `$KC_HOME/data/password-blacklists/` by default. Customize this path using:
+** The `keycloak.password.blacklists.path` system property.
+** The `blacklistsPath` property of the `passwordBlacklist` policy SPI configuration. To configure the blacklist folder via CLI use `--spi-password-policy-password-blacklist-blacklists-path=/path/to/blacklistsFolder`.
+
+A note about False Positives
+
+The current implementation uses a BloomFilter for fast and memory efficient containment checks, e.g. whether a given password is contained in a blacklist, with the possibility for false positives. 
+
+* By default a false positive probability of `0.01%` is used.
+* To change the false positive probability via CLI configuration use `--spi-password-policy-password-blacklist-false-positive-probability=0.00001`
```
Comment (andymunro) [inline server_admin/topics/authentication/password-policies.adoc]:
```suggestion
* To change the false positive probability by CLI configuration, use `--spi-password-policy-password-blacklist-false-positive-probability=0.00001`
```

**Review by andymunro — CHANGES_REQUESTED** [review]
A few suggested changes.

**Comment by thomasdarimont** [comment]
@andymunro thanks for your review! I just applied your suggestions and rebased the branch on main.

**Comment by thomasdarimont** [comment]
I also fixed the three broken links (http -> https)  that caused this PR to fail.

**Comment by ahus1** [comment]
@thomasdarimont - thank you for providing this documentation update. 

Due to our long feedback cycle, this acquired some conflicts. Please rebase this PR. 

The one link to CSP has been fixed in the main branch already. 

The other change in the change log was also fixed, but differently from your suggestion: It was actually a key to be placed in the configuration file, not a link to be clicked, therefore it has been marked as pass-though content. 

Thanks!

**UPDATE:** Please also squash your commits. Thanks!

**Comment by ahus1** [comment]
Superseded by https://github.com/keycloak/keycloak/pull/19280

**Pull request closed without merging**